source/opencascade/utilities.cc

```diff
@@ -267,6 +275,70 @@ namespace OpenCASCADE
     AssertThrow(OK, ExcMessage("Failed to write IGES file."));
   }
 
+
+  TopoDS_Shape
+  read_STL(const std::string &filename)
+  {
+    StlAPI_Reader reader;
+    TopoDS_Shape  shape;
+    reader.Read(shape, filename.c_str());
+    return shape;
+  }
+
+
+  void
+  write_STL(const TopoDS_Shape &shape,
+            const std::string & filename,
+            const double        deflection,
+            const bool          sew_different_faces,
+            const double        sewer_tolerance,
+            const bool          is_relative,
+            const double        angular_deflection,
+            const bool          in_parallel)
+  {
+    TopLoc_Location            Loc;
+    std::vector<TopoDS_Vertex> vertices;
+    std::vector<TopoDS_Edge>   edges;
+    std::vector<TopoDS_Face>   faces;
+    OpenCASCADE::extract_geometrical_shapes(shape, faces, edges, vertices);
+    bool mesh_is_present = true;
+    for (unsigned int i = 0; i < faces.size(); ++i)
+      {
+        Handle(Poly_Triangulation) theTriangulation =
+          BRep_Tool::Triangulation(faces[i], Loc);
+        if (theTriangulation.IsNull())
+          {
+            mesh_is_present = false;
+            break;
+          }
+      }
+    TopoDS_Shape shape_to_be_written = shape;
+    if (!mesh_is_present)
+      {
+        if (sew_different_faces)
+          {
+            BRepBuilderAPI_Sewing sewer(sewer_tolerance);
+            sewer.Add(shape_to_be_written);
+            sewer.Perform();
+            shape_to_be_written = sewer.SewedShape();
+          }
+        else
+          shape_to_be_written = shape;
+        BRepMesh_IncrementalMesh mesh_im(shape_to_be_written,
+                                         deflection,
+                                         is_relative,
```

Comment:
I am confused by this statement. You are creating an object of type `BRepMesh_IncrementalMesh` here in line 327, but this is the last statement in the block -- the next line after the creation of the object is a closing brace. So the object is dead -- it is destroyed again immediately after being created. What is it there for?

Reply:
Occ automatically calls a perform method that creates the triangulation in the shape.

Reply:
So one of the constructor arguments here is an output argument? Can you please add a summary of this discussion as a comment so that it's clear to the next reader how this work? -- Thanks!

Reply:
Ok, done